### PR TITLE
install: Deploy peerpod-ctrl by default

### PR DESCRIPTION
**Pull request by bpradipt**
The default value of RESOURCE_CTRL is set to true to allow deletion of dangling resources.

### CODE REVIEW

**Review by huoqifeng — APPROVED** [review]
LGTM

**Comment by huoqifeng** [comment]
Thank you @bpradipt

**Review by stevenhorsman — COMMENTED** [review]
Do we need to some something to make the e2e tests also deploy the peerpod-ctrl?
Maybe after https://github.com/confidential-containers/cloud-api-adaptor/blob/827d0f1b133de69a220147d0c3e801921e2731ef/test/provisioner/provision.go#L218-L221? I want to avoid us having different default config in the CI tests from the other deployment approach(es). 

Maybe we should add a line to the manual install path of the readme as well to tie up those loose ends?

**Comment by bpradipt** [comment]
> Do we need to some something to make the e2e tests also deploy the peerpod-ctrl? Maybe after
> 
> https://github.com/confidential-containers/cloud-api-adaptor/blob/827d0f1b133de69a220147d0c3e801921e2731ef/test/provisioner/provision.go#L218-L221
> 
> ? I want to avoid us having different default config in the CI tests from the other deployment approach(es).
> Maybe we should add a line to the manual install path of the readme as well to tie up those loose ends?

Yeah good point.  I'll look into how to enable it in e2e by default

**Review by stevenhorsman — COMMENTED** [review]
The provisioning update code looks okay to me. Sorry to be a pain, but should we consider running peerpod-ctrl undeploy somewhere around https://github.com/confidential-containers/cloud-api-adaptor/blob/1b21353db874d12c8fdc22295ab68ded987ce43a/test/provisioner/provision.go#L132?

File: test/provisioner/provision.go [hunk 1]
```diff
@@ -256,13 +279,34 @@ func (p *CloudAPIAdaptor) Deploy(ctx context.Context, cfg *envconf.Config, props
...
+	}
+
 	return nil
 }
 
 func (p *CloudAPIAdaptor) DoKustomize(ctx context.Context, cfg *envconf.Config) {
 }
 
-// TODO: convert this into a klient/wait/conditions
+// TODO: convert this into a client/wait/conditions
```
Comment (stevenhorsman) [inline test/provisioner/provision.go:309]: `sigs.k8s.io/e2e-framework/klient` is the go package that waits and conditions are under, so I don't think this was a typo

Comment (bpradipt) [reply test/provisioner/provision.go:309]: Ah this was autocorrect on save :-( .. Need to disable on my new editor setup

**Review by stevenhorsman — APPROVED** [review]
Code LGTM, so as long as the e2e libvirt tests pass, I'm happy for this to be merged. Thanks @bpradipt for the code and updates.

**Comment by stevenhorsman** [comment]
The e2e tests are failing with:
```
Traceback (most recent call last):
  File "/usr/bin/kcli", line 33, in <module>
    sys.exit(load_entry_point('kcli==99.0', 'console_scripts', 'kcli')())
  File "/usr/lib/python3/dist-packages/kvirt/cli.py", line 5416, in cli
    args.func(args)
  File "/usr/lib/python3/dist-packages/kvirt/cli.py", line 131, in get_version
    git_version, git_date = get_git_version()
  File "/usr/lib/python3/dist-packages/kvirt/common/__init__.py", line [21](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/7570170726/job/20616589992?pr=1665#step:10:22)14, in get_git_version
    git_version, git_date = open(git_file).read().rstrip().split(' ')
```
this looks like the same issue that @karmab fixed a couple of weeks ago, though 😕

**Comment by wainersm** [comment]
hi @bpradipt !

In order to test peerpod-ctrl changes on pull request you will need to build and push its image like we do for podvm and cloud-api-adaptor images. Otherwise it will be testing `quay.io/confidential-containers/peerpod-ctrl:latest`, that is [published on every merge](https://github.com/confidential-containers/cloud-api-adaptor/actions/workflows/publish_images_on_push.yaml).

**Comment by karmab** [comment]
> The e2e tests are failing with:
> 
> ```
> Traceback (most recent call last):
>   File "/usr/bin/kcli", line 33, in <module>
>     sys.exit(load_entry_point('kcli==99.0', 'console_scripts', 'kcli')())
>   File "/usr/lib/python3/dist-packages/kvirt/cli.py", line 5416, in cli
>     args.func(args)
>   File "/usr/lib/python3/dist-packages/kvirt/cli.py", line 131, in get_version
>     git_version, git_date = get_git_version()
>   File "/usr/lib/python3/dist-packages/kvirt/common/__init__.py", line [21](https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/7570170726/job/20616589992?pr=1665#step:10:22)14, in get_git_version
>     git_version, git_date = open(git_file).read().rstrip().split(' ')
> ```
> 
> this looks like the same issue that @karmab fixed a couple of weeks ago, though 😕

let me check

**Comment by bpradipt** [comment]
> hi @bpradipt !
> 
> In order to test peerpod-ctrl changes on pull request you will need to build and push its image like we do for podvm and cloud-api-adaptor images. Otherwise it will be testing `quay.io/confidential-containers/peerpod-ctrl:latest`, that is [published on every merge](https://github.com/confidential-containers/cloud-api-adaptor/actions/workflows/publish_images_on_push.yaml).

@wainersm that's not the intent and frankly out of scope for this PR. peerpod-ctrl is treated like the operator here.
Sometime back we had a discussion to move out peerpod-ctrl and peerpodconfig-ctrl to it's own repos to make it easier for testing.

**Comment by wainersm** [comment]
> > hi @bpradipt !
> > In order to test peerpod-ctrl changes on pull request you will need to build and push its image like we do for podvm and cloud-api-adaptor images. Otherwise it will be testing `quay.io/confidential-containers/peerpod-ctrl:latest`, that is [published on every merge](https://github.com/confidential-containers/cloud-api-adaptor/actions/workflows/publish_images_on_push.yaml).
> 
> @wainersm that's not the intent and frankly out of scope for this PR. peerpod-ctrl is treated like the operator here. Sometime back we had a discussion to move out peerpod-ctrl and peerpodconfig-ctrl to it's own repos to make it easier for testing.

Alright, thanks for the explanation!

**Pull request merged**